### DISCUSSION TITLE
There are version incompatibilities between Gauge and it's plugins in this project. Some features will not work as expected.

### DISCUSSION THREAD

**Discussion by GavinSun9527** [getgauge/gauge#2523] [General]
we have a gauge test project and upgrade from netframework to .net 8, and we want to run & debug the gauge in vs code, but I got this message when I open the project with vs code.

my vs code gauge extension version: 0.1.6
My Environment Information -----------
	windows, 1.4.3, f98dd40
	csharp (0.10.6), dotnet (0.5.7), html-report (4.3.1), screenshot (0.2.0), xml-report (0.5.1)

does anyone knows how check the version issue?

**Comment by chadlwilson** [comment]
Same as https://github.com/getgauge/gauge-vscode/issues/945  ?

**Reply by GavinSun9527** [reply]
yes
I have updated to Gauge 1.6.6, but it still has the same issue

and I tried click run Scenario , I will got these message from output

Running tool: gauge.cmd run --hide-suggestion --simple-console c:\xxxxxxxxx (mark)\ProcessPaymentTests.spec:9
Building project: c:\xxxxxxxxxxxxxx.Gauge.Tests.csproj
c:\Windows\Microsoft.NET\Framework\v4.0.30319\MSBuild.exe c:\xxxxxxxxxxxx.Tests.csproj /t:Build /m /nodeReuse:False /v:q /p:RestorePackages="False" /p:OutputPath="c:\xxxxxxxxxx\gauge_bin" /p:Platform="Any CPU" /p:Configuration="Debug" /logger:Fake.MsBuildLogger+ErrorLogger,"C:\Users\gsun\AppData\Roaming\gauge\plugins\csharp\0.10.6\bin\FakeLib.dll"
Microsoft (R) Build Engine version 4.8.9037.0
[Microsoft .NET Framework, version 4.0.30319.42000]
Copyright (C) Microsoft Corporation. All rights reserved.
c:\xxxxxxxxxxxxxxxxx.Gauge.Tests.csproj(1,1): error MSB4041: The default XML namespace of the project must be the MSBuild XML namespace. If the project is authored in the MSBuild 2003 f

do you know why it is still built with Microsoft.NET\Framework\v4.0.30319\MSBuild.exe ？ I have upgraded to .NET 8 for my project

**Reply by chadlwilson** [reply]
Please don't open/discuss the same issue in two different places. Let's stick to one place please.

**Reply by GavinSun9527** [reply]
sure,let's discuss here and let me close another the issue

**Comment by GavinSun9527** [comment]
Hi @chadlwilson 
And another case is , I have another gauge test project, and I upgrade from NET Stardand 2.1 to .net 8,  before upgrade all test cases can run success in vs code.

After upgrade (and upgrade the gauge version) , almost spec are failed

fail: Grpc.AspNetCore.Server.ServerCallHandler[6]
Error when executing service method 'ValidateStep'.
System.InvalidOperationException: Sequence contains no matching element
at System.Linq.ThrowHelper.ThrowNoMatchException()
at System.Linq.Enumerable.First[TSource](IEnumerable`1 source, Func`2 predicate)
at Gauge.Dotnet.AssemblyLoader.GetMethods(LibType type) in /home/runner/work/gauge-dotnet/gauge-dotnet/src/AssemblyLoader.cs:line 53
at Gauge.Dotnet.Models.HookRegistry..ctor(IAssemblyLoader assemblyLoader) in /home/runner/work/gauge-dotnet/gauge-dotnet/src/Models/HookRegistry.cs:line 42
at Gauge.Dotnet.HookExecutor..ctor(IAssemblyLoader assemblyLoader, IReflectionWrapper reflectionWrapper, Object classInstanceManager, IExecutionInfoMapper mapper) in /home/runner/work/gauge-dotnet/gauge-dotnet/src/HookExecutor.cs:line 30


And this is a failed spec file content, Each step (start with *) line has a red wavy underline

```
# Customer Tests
* Set "Sandbox" As Current Merchant

------------------------------------------------- Post Validation Start ---------------------------------------------------
## Run multiple create customer successfully

tags: testCustomer

   |PropertyName|PropertyValue|
   |------------|-------------|
   |reference   |[NEWID]      |
   |reference   |             |
   |email       |             |
* Generate and send default Create Customer Request for the "test" Merchant with Overriden Values

   |Property      |Value          |
   |--------------|---------------|
   |<PropertyName>|<PropertyValue>|
* Verify Create Customer Response Successful With ResultStatus "ok"
```

**Reply by chadlwilson** [reply]
Ok, you seem to be talking about multiple different things now, there is dotnet support handled by the gauge-dotnet plugin and vscode integration with the gauge-vscode plugin. Lets try and address one thing at a time.

I'm not familiar with the dotnet plugin, it may need work to support dotnet 8. I know specific work was done to support dotnet 7, so maybe we can eliminate that plugin as the problem by trying with dotnet 7 first and make sure both vscode, your code and the dotnet plugin are working ok.

Perhaps simplify your environment by removing the gauge-csharp plugin - perhaps gauge is still trying to use that, which won't work with .net core, and is replaced by the gauge-dotnet support for .net core.

**Reply by GavinSun9527** [reply]
I tried rollback the gauge project to net stardand 2.1 and re-install gauge 1.4.3 , but still has this issue

**Reply by chadlwilson** [reply]
Sounds like you are trying to change too many things at once and there is something wrong with your environment.

Since we can't see the code changes you are also making to try and migrate dotnet versions it's difficult to assist. Dotnet core 2.1 has been EOL for many years now, so I suspect you'll have difficulty getting help specific to that.

**Reply by GavinSun9527** [reply]
it may caused by the old plugins , I uninstalled all plugins and re-install again, and it worked for the netstandard 2.1

**Reply by chadlwilson** [reply]
Ok, so now you will try with dotnet core 7 perhaps?

**Reply by GavinSun9527** [reply]
I will try but it may take some time,  so another question is my another gauge project, it upgrdge from .NET framework 4.8 to .NET 8, and in vs2022 , it builds successful, but in vs code when I click run scenario, it will output the error message (see the details at the beginning of this discussions)

do you know why it is still built with Microsoft.NET\Framework\v4.0.30319\MSBuild.exe ？

**Reply by chadlwilson** [reply]
As I said before, maybe because you still have the gauge-csharp plugin installed and it is using that, or maybe because you haven't correctly converted your project to dotnet core, but I don't know.

**Reply by GavinSun9527** [reply]
Sure, I got your point, but in vs code "TERMINAL" tab, if I run the command "dotnet build -c Release" in the project folder path, it can build success
like as the follows
```
MSBuild version 17.8.3+195e7f5a3 for .NET
  Determining projects to restore...
  All projects are up-to-date for restore.
...
```

**Reply by chadlwilson** [reply]
Did you remove the dotnet-csharp plugin and see if it's any different?

**Reply by GavinSun9527** [reply]
Okay, let me try

today I got these message from vs code output when I open the gauge project
```
[Error - 10:50:00 AM] An error occurred. Refer lsp.log for more details.
[Info  - 10:50:00 AM] Connection to server got closed. Server will restart.
true
[Error - 10:50:02 AM] An error occurred. Refer lsp.log for more details.
[Info  - 10:50:02 AM] Connection to server got closed. Server will restart.
true
[Error - 10:50:03 AM] An error occurred. Refer lsp.log for more details.
[Info  - 10:50:03 AM] Connection to server got closed. Server will restart.
true

```

check the lsp.log
```
15-04-2024 10:50:00.957 [Gauge] [INFO] Starting language runner
15-04-2024 10:50:01.813 [dotnet] [INFO] MSBuild version 17.8.3+195e7f5a3 for .NET
15-04-2024 10:50:01.824 [dotnet] [INFO] MSBUILD : error MSB1006: Property is not valid.
15-04-2024 10:50:01.824 [dotnet] [INFO] Switch:  LLC\Desktop\TestUpgrade\eWAY.GlobalRapidAPI.Gauge.Tests\gauge_bin
15-04-2024 10:50:01.824 [dotnet] [INFO] For switch syntax, type "MSBuild -help"
15-04-2024 10:50:02.287 [Gauge] [INFO] LangServer: reading on stdin, writing on stdout
15-04-2024 10:50:02.290 [Gauge] [CRITICAL] notif #0: initialized: runtime error: invalid memory address or nil pointer dereference
goroutine 26 [running]:
runtime/debug.Stack(0xc00065d780, 0x1261480, 0x1822a30)
	C:/hostedtoolcache/windows/go/1.16.13/x64/src/runtime/debug/stack.go:24 +0xa5
github.com/getgauge/gauge/api/lang.recoverPanic(0xc000154550)
	D:/a/gauge/gauge/api/lang/server.go:308 +0x65
panic(0x1261480, 0x1822a30)
	C:/hostedtoolcache/windows/go/1.16.13/x64/src/runtime/panic.go:965 +0x1c7
github.com/getgauge/gauge/runner.(*GrpcRunner).ExecuteMessageWithTimeout(...)
	D:/a/gauge/gauge/runner/grpcRunner.go:195
github.com/getgauge/gauge/api/lang.globPatternRequest(0x4, 0x4, 0x1205aa0)
	D:/a/gauge/gauge/api/lang/runner.go:87 +0x7e
github.com/getgauge/gauge/api/lang.registerRunnerCapabilities(0x1431c40, 0xc0000ce000, 0x14347b0, 0xc0000a6088, 0x0, 0x0)
	D:/a/gauge/gauge/api/lang/capabilities.go:110 +0x54
github.com/getgauge/gauge/api/lang.(*LangHandler).Handle(0x18803c8, 0x14347b0, 0xc0000a6088, 0x1431c40, 0xc0000ce000, 0xc000154550, 0xc00000bf08, 0xb36066, 0xc00008e1e0, 0xc0001648a0)
	D:/a/gauge/gauge/api/lang/server.go:76 +0x649
github.com/getgauge/gauge/api/lang.(*LangHandler).handle(0x18803c8, 0x14347b0, 0xc0000a6088, 0xc0000ce000, 0xc000154550, 0x0, 0x0, 0x0, 0x0)
	D:/a/gauge/gauge/api/lang/server.go:60 +0xac
github.com/sourcegraph/jsonrpc2.(*HandlerWithErrorConfigurer).Handle(0xc000508120, 0x14347b0, 0xc0000a6088, 0xc0000ce000, 0xc000154550)
	C:/Users/runneradmin/go/pkg/mod/github.com/sourcegraph/jsonrpc2@v0.0.0-20200429184054-15c2290dcb37/handler_with_error.go:21 +0x79
created by github.com/getgauge/gauge/api/lang.lspHandler.Handle
	D:/a/gauge/gauge/api/lang/server.go:55 +0x72

```

I tried to create a new gauge project in vs code, but still get this error message, do you know what happened？

**Reply by chadlwilson** [reply]
I don't know, but it looks like the vscode plugin is not talking correctly to the gauge server. (the `LSP` is the bit that understands the language syntax and works with VSCode to make VSCode Gauge-aware) 

I would make sure you are on latest/recent versions of all the pieces e.g VSCode `1.88.1`, Gauge `1.6.6`, gauge-vscode `0.1.6` and gauge-dotnet `0.5.7`. That *looks* like an old version of Gauge to me.
